R/data.table.R

```diff
@@ -49,6 +49,32 @@ null.data.table = function() {
   setalloccol(ans)
 }
 
+rowwiseDT = function(..., key=NULL) {
+  x = substitute(list(...))[-1L]
+  if (is.null(nms <- names(x)))
+    stop("Must provide at least one column (use `name=`)")
+  header_pos = which(nzchar(nms))
+  if (any(nzchar(x[header_pos])))
+    stop("Named arguments must be empty")
+  if (!identical(header_pos, seq_along(header_pos)))
+    stop("Header must be the first N arguments")
+  header = nms[header_pos]
+  ncols = length(header)
+  body = lapply(x[-header_pos], eval, envir = parent.frame())
+  if (length(body) %% ncols != 0L) 
+    stop(sprintf("There're %d columns but the number of cells is %d, which is not an integer multiple of the columns", ncols, length(body)))
+  # make all the non-scalar elements to a list
+  body = lapply(body, function(x) if (length(x) != 1L) list(x) else x)
+  body = split(body, rep(seq_len(length(body) / ncols), each = ncols))
+  ans = rbindlist(body)
+  setnames(ans, header)
+  if (!is.null(key)) {
```

Comment:
Is the ```key``` arg needed for rowwise? It just seems like most uses would be such small tables that I doubt the ```key``` argument would get any appreciable use. Plus, setting a key is a single line that would likely be suited on a separate line.
```
dt = rowwiseDT(...)
setDT(dt, "something")
```